Audacia.Quest/Audacia.Quest.Core/Events/EventsCollection.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Audacia.Quest.Core
+{
+    public class EventsCollection
+    {
+        private static IDictionary<string, object> AvailableEvents { get; set; } = new Dictionary<string, object>();
+
+        public static void Add<T>(string name, IEvent<T> e)
+        {
+            if (!AvailableEvents.ContainsKey(name))
+            {
+                AvailableEvents.Add(name, e);
+            }
+        }
+
+        public static void PerformEvent<T>(string name, T param)
```

Comment:
I would suggest to have some validation here in checking whether or not the event within the `AvailableEvents` as currently I think it could throw null exceptions if unable to find the key